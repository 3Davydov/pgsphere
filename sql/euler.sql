\set ECHO none
SELECT set_sphere_output_precision(8);
SET extra_float_digits TO -3;
\set ECHO all

-- checking Euler transformation operators

SELECT strans '-10d,0d,10d,ZZZ'  = '-10d,0d,10d,XXX' ;
SELECT strans '-40d,0d,40d,ZZZ' <> '-40d,0d,40d,XXX' ;

SELECT strans ( 20.0*pi()/180.0, -270.0*pi()/180.0, 70.5*pi()/180.0, 'XZY' );

SELECT theta( strans ( 20.0*pi()/180.0, -270.0*pi()/180.0, 70.5*pi()/180.0, 'XZY' ) );

SELECT psi( strans ( 20.0*pi()/180.0, -270.0*pi()/180.0, 70.5*pi()/180.0, 'XZY' ) );

SELECT phi( strans ( 20.0*pi()/180.0, -270.0*pi()/180.0, 70.5*pi()/180.0, 'XZY' ) );

SELECT theta( strans( '20d, 30d, 40d, XZY' ) );

SELECT psi( strans( '20d, 30d, 40d, XZY' ) );

SELECT phi( strans( '20d, 30d, 40d, XZY' ) );

SELECT strans( '2d 20m, 10d, 0' );

SELECT theta( strans( '2d 20m, 10d, 0' ) );

SELECT psi( strans( '2d 20m, 10d, 0' ) );

SELECT phi( strans( '2d 20m, 10d, 0' ) );

SELECT strans ( '10d, 90d, 270d, ZXZ' );

SELECT theta( strans ( '10d, 90d, 270d, ZXZ' ) );

SELECT psi( strans ( '10d, 90d, 270d, ZXZ' ) );

SELECT phi( strans ( '10d, 90d, 270d, ZXZ' ) );

SELECT - strans ( '20d, 50d, 80d, XYZ' );

SELECT theta( - strans ( '20d, 50d, 80d, XYZ' ) );

SELECT psi( - strans ( '20d, 50d, 80d, XYZ' ) );

SELECT phi( - strans ( '20d, 50d, 80d, XYZ' ) );

SELECT strans( '90d, 60d, 30d' );

SELECT theta( strans( '90d, 60d, 30d' ) );

SELECT psi( strans( '90d, 60d, 30d' ) );

SELECT phi( strans( '90d, 60d, 30d' ) );
