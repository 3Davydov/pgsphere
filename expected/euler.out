\set ECHO none
 set_sphere_output_precision 
-----------------------------
 SET 8
(1 row)

-- checking Euler transformation operators
SELECT strans '-10d,0d,10d,ZZZ'  = '-10d,0d,10d,XXX' ;
 ?column? 
----------
 t
(1 row)

SELECT strans '-40d,0d,40d,ZZZ' <> '-40d,0d,40d,XXX' ;
 ?column? 
----------
 f
(1 row)

SELECT strans ( 20.0*pi()/180.0, -270.0*pi()/180.0, 70.5*pi()/180.0, 'XZY' );
                strans                 
---------------------------------------
 0.34906585, 1.5707963, 1.2304571, XZY
(1 row)

SELECT theta( strans ( 20.0*pi()/180.0, -270.0*pi()/180.0, 70.5*pi()/180.0, 'XZY' ) );
     theta     
---------------
 1.57079632679
(1 row)

SELECT psi( strans ( 20.0*pi()/180.0, -270.0*pi()/180.0, 70.5*pi()/180.0, 'XZY' ) );
      psi      
---------------
 1.23045712266
(1 row)

SELECT phi( strans ( 20.0*pi()/180.0, -270.0*pi()/180.0, 70.5*pi()/180.0, 'XZY' ) );
      phi       
----------------
 0.349065850399
(1 row)

SELECT theta( strans( '20d, 30d, 40d, XZY' ) );
     theta      
----------------
 0.523598775598
(1 row)

SELECT psi( strans( '20d, 30d, 40d, XZY' ) );
      psi       
----------------
 0.698131700798
(1 row)

SELECT phi( strans( '20d, 30d, 40d, XZY' ) );
      phi       
----------------
 0.349065850399
(1 row)

SELECT strans( '2d 20m, 10d, 0' );
             strans              
---------------------------------
 0.040724349, 0.17453293, 0, ZXZ
(1 row)

SELECT theta( strans( '2d 20m, 10d, 0' ) );
     theta      
----------------
 0.174532925199
(1 row)

SELECT psi( strans( '2d 20m, 10d, 0' ) );
 psi 
-----
   0
(1 row)

SELECT phi( strans( '2d 20m, 10d, 0' ) );
       phi       
-----------------
 0.0407243492132
(1 row)

SELECT strans ( '10d, 90d, 270d, ZXZ' );
                strans                
--------------------------------------
 0.17453293, 1.5707963, 4.712389, ZXZ
(1 row)

SELECT theta( strans ( '10d, 90d, 270d, ZXZ' ) );
     theta     
---------------
 1.57079632679
(1 row)

SELECT psi( strans ( '10d, 90d, 270d, ZXZ' ) );
      psi      
---------------
 4.71238898038
(1 row)

SELECT phi( strans ( '10d, 90d, 270d, ZXZ' ) );
      phi       
----------------
 0.174532925199
(1 row)

SELECT - strans ( '20d, 50d, 80d, XYZ' );
               ?column?               
--------------------------------------
 4.8869219, 5.4105207, 5.9341195, ZYX
(1 row)

SELECT theta( - strans ( '20d, 50d, 80d, XYZ' ) );
      theta      
-----------------
 -0.872664625997
(1 row)

SELECT psi( - strans ( '20d, 50d, 80d, XYZ' ) );
       psi       
-----------------
 -0.349065850399
(1 row)

SELECT phi( - strans ( '20d, 50d, 80d, XYZ' ) );
      phi      
---------------
 -1.3962634016
(1 row)

SELECT strans( '90d, 60d, 30d' );
                strans                 
---------------------------------------
 1.5707963, 1.0471976, 0.52359878, ZXZ
(1 row)

SELECT theta( strans( '90d, 60d, 30d' ) );
    theta     
--------------
 1.0471975512
(1 row)

SELECT psi( strans( '90d, 60d, 30d' ) );
      psi       
----------------
 0.523598775598
(1 row)

SELECT phi( strans( '90d, 60d, 30d' ) );
      phi      
---------------
 1.57079632679
(1 row)

